### PR TITLE
iproute2: fix build with ip-tiny deselected, reapply reverted commits

### PR DESCRIPTION
**Pull request by guidosarducci**
**Maintainers/Distribution List**: @dedeckeh  @ldir-EDB0
**Compile tested**: mips32-be-malta, master branch (test all selection combinations of ip-tiny, ip-full, tc)
**Run tested**: mips32-be-malta,  master branch
**Related PRs**: #1627 (merged, partially reverted)

**Description:**
This follows up from the latest discussion in #1627, and simply drops attempted fixes to work around inconsistent build products, which caused its own build problems.

Please review, and thanks.

### CODE REVIEW

File: package/network/utils/iproute2/Makefile [hunk 1]
```diff
@@ -127,12 +130,15 @@ define Build/Compile
 	+$(MAKE_VARS) $(MAKE) $(PKG_JOBS) -C $(PKG_BUILD_DIR) $(MAKE_FLAGS)
 endef
 
+# use minimal functionality for dev libraries
+ifeq ($(BUILD_VARIANT),tiny)
```
Comment (dedeckeh) [inline package/network/utils/iproute2/Makefile]: This will break user packages linking to libnetlink.a when ip-full is selected

Comment (guidosarducci) [reply package/network/utils/iproute2/Makefile]: It seems you're not looking at the branch tip but reviewing outdated code 1 commit behind; please check again.

Comment (ldir-EDB0) [reply package/network/utils/iproute2/Makefile]: @guidosarducci I think a lack of clarity from myself here.  Ideally each commit in the master tree should build without error.  Having a commit still in that 'breaks' stuff followed by a commit that fixes the break isn't ideal.  If you squash the fix commit into the corresponding prior commit so nothing 'breaks' on the way then I'm sure that would be better.

Comment (guidosarducci) [reply package/network/utils/iproute2/Makefile]: This is simply the PR which you requested yourself and @ldir-EDB0 confirmed in https://github.com/openwrt/openwrt/pull/1627#issuecomment-465466958 and https://github.com/openwrt/openwrt/pull/1627#issuecomment-465489270 respectively:
>I'm fine opening a separate PR which re-applies the reverted commits together with the fixes

and
>I agree that a separate PR with the 4 reverted commits and the fix commit would be a fair way forward.

The branch tip **is** the "fix commit", and doesn't break iproute2 as you claim. The same commit could have been cleanly and quickly applied to master without a mass reversion, thus avoiding unneeded churn.

File: package/network/utils/iproute2/Makefile [hunk 2]
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=iproute2
 PKG_VERSION:=4.20.0
-PKG_RELEASE:=1
+PKG_RELEASE:=3
```
Comment (ldir-EDB0) [inline package/network/utils/iproute2/Makefile]: Any reason why 3?  Typo?

**Comment by dedeckeh** [comment]
Patches pushed to master (https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=f61359e16ef911a8fc01e7028942d9cad0bd8a96, https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=0b57a2165a4c9c0f5f5032d7924ef65539571b1f, https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=b61495409bb398b1967d7da305a49a6f077cb55a); thx

**Pull request closed without merging**